Cli-CredentialHelper/Program.cs

```diff
@@ -776,8 +779,123 @@ private static void EnableTraceLogging(OperationArguments operationArguments)
+                                                                              ref saveCredentials,
+                                                                              flags)) != NativeMethods.Win32Error.Success)
+                {
+                    username = null;
+                    password = null;
+
+                    Trace.WriteLine("   credential prompt failed (" + NativeMethods.Win32Error.GetText(error) + ").");
+
+                    return false;
+                }
+
+                // use `StringBuilder` references instead of string so that they can be written to
+                StringBuilder usernameBuffer = new StringBuilder(512);
+                StringBuilder domainBuffer = new StringBuilder(256);
+                StringBuilder passwordBuffer = new StringBuilder(512);
+                int usernameLen = usernameBuffer.Capacity;
+                int passwordLen = passwordBuffer.Capacity;
+                int domainLen = domainBuffer.Capacity;
+
+                // unpack the result into locally useful data
+                if (!NativeMethods.CredUnPackAuthenticationBuffer(authPackage,
+                                                                  packedAuthBufferPtr,
+                                                                  packedAuthBufferSize,
+                                                                  usernameBuffer,
+                                                                  ref usernameLen,
+                                                                  domainBuffer,
+                                                                  ref domainLen,
+                                                                  passwordBuffer,
+                                                                  ref passwordLen))
+                {
+                    username = null;
+                    password = null;
+
+                    error = Marshal.GetLastWin32Error();
+                    Trace.WriteLine("   failed to unpack buffer (" + NativeMethods.Win32Error.GetText(error) + ").");
+
+                    return false;
+                }
+
+                username = usernameBuffer.ToString();
+                password = passwordBuffer.ToString();
+
+                return true;
+            }
+            catch { throw; }
+            finally
+            {
+                if (packedAuthBufferPtr != IntPtr.Zero)
+                {
+                    Marshal.FreeHGlobal(packedAuthBufferPtr);
+                }
+            }
+        }
+
+        private static bool GithubCredentialModalPrompt( Uri targetUri, out string username, out string password ) {
+            return PromptForCredentials( targetUri, out username, out password );
+        }
+
+        private static bool GithubAuthCodeModalPrompt( Uri targetUri, GithubAuthenticationResultType resultType, out string authenticationCode ) {
```

Comment:
This is novel, but it will be unclear to most users what they should place in the username and password fields.

Maybe it would better to pre-populate the username with the previously provided value and ask for the auth code in the password field?